examples/pennylane_run_rotoselect.py

```diff
@@ -0,0 +1,405 @@
+#
+# VQE
+# ~~~
+#
+# We choose to focus on the example of a VQE circuit with 2 qubits for simplicity. Here, the Hamiltonian
+# is
+#
+# .. math::
+#   H = 0.5Y_2 - 0.8Z_1 - 0.2X_1
+#
+# where the subscript denotes the qubit upon which the Pauli operator acts. The
+# expectation value of this quantity acts as the cost function for our
+# optimization.
+#
+# Rotosolve
+# ---------
+# As a precursor to implementing Rotoselect we can analyze a version of the algorithm
+# which does not optimize the choice of gates, called Rotosolve. Later, we will build on this example
+# to implement Rotoselect and vary the circuit structure.
+#
+# Imports
+# ~~~~~~~
+# To get started, we import PennyLane and the PennyLane-included version of NumPy. We also
+# create a 2-qubit device using the ``default.qubit`` plugin and set the ``analytic`` keyword to ``True``
+# in order to obtain exact values for any expectation values calculated. In contrast with real
+# devices, simulators have the capability of doing these calculations without sampling.
+
+import pennylane as qml
+from pennylane import numpy as np
+
+n_wires = 2
+
+dev = qml.device('default.qubit',analytic=True,wires=2)
+
+##############################################################################
+# Creating a fixed quantum circuit
+# ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+#
+# .. figure:: ../../examples/figures/original_ansatz.png
+#    :scale: 65%
+#    :align: center
+#    :alt: original_ansatz
+#
+# |
+# Next, we set up a circuit with a fixed ansatz structure -- which will later be subject to change -- and encode
+# the Hamiltonian into a cost function. The structure https://github.com/XanaduAI/pennylane/pull/335is shown in the figure above.
+
+def ansatz(params):
+    qml.RX(params[0], wires=0)
+    qml.RY(params[1], wires=1)
+    qml.CNOT(wires=[0, 1])
+
+
+@qml.qnode(dev)
+def circuit(params):
+    ansatz(params)
+    return qml.expval(qml.PauliZ(0)),qml.expval(qml.PauliY(1))
+
+@qml.qnode(dev)
+def circuit2(params):
+    ansatz(params)
+    return qml.expval(qml.PauliX(0))
+
+def cost(params):
+    Z_1 = circuit(params)[0]
+    Y_2 = circuit(params)[1]
+    X_1 = circuit2(params)
+    return 0.5 * Y_2 + 0.8 * Z_1 - 0.2 * X_1
+
+##############################################################################
+# Helper methods for the algorithm
+# ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+# We define methods to evaluate
+# the expression in the previous section. These will serve as the basis for
+# our optimization algorithm.
+
+# calculation as described above
+def opt_theta_d(d, params, cost):
+    params[d] = 0.
+    M_0 = cost(params)
+    params[d] = np.pi / 2.
+    M_0_plus = cost(params)
+    params[d] = -np.pi / 2.
+    M_0_minus = cost(params)
+    a = np.arctan2(2. * M_0 - M_0_plus - M_0_minus, M_0_plus - M_0_minus)
+    params[d] = -np.pi / 2. - a
+    # restrict output to lie in (-pi,pi], a convention
+    # consistent with the Rotosolve paper
+    if params[d] > np.pi:
```

Comment:
Ah, I should probably mention that the numpy function arctan2 returns a value in range (-pi,pi] so what I've done is sufficient and in fact overkill since I only need to address the case where the return value is less than -pi/2. I'll state this in a new comment and remove the first unnecessary if statement.